hack/components/bump-knmstate.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+set -xeo pipefail
+
+source hack/components/yaml-utils.sh
+source hack/components/git-utils.sh
+
+NMSTATE_URL=$(yaml-utils::get_component_url nmstate)
```

Comment:
The usage of sub-shelling here is strange.
- If you sub-shell, then you could just do `./hack/components/yaml-utils.sh --get-component-url nmstate` and there is no need for sourcing. The advantage of sourcing as I see it, is the avoidance of sub-shelling.
- Another alternative is for the function to return the output and sub-shell if needed there.

I lost track in this review of what existed and what was actually added, so if it was already like this before, feel free to do this in a following PR.

Reply:
not sure what you mean by strange. can you elaborate what's wrong with it, so that I could unserstand why your suggestions are better for this situation?

Reply:
If you sub-shell, you do not need function calls, you just need to call the script.
I would expect with `$()` to see shell commands, not calls to functions: `$(ls -la)`.

Reply:
I don't know about performance cost, but when it comes to readability, I am a fan of subshell called on sourced functions.

It allows us to do something close to imports and calling functions from scopes. The alternative would be to either have one shell script per function or to keep a boilerplate to select the right function based on an attribute.

Is there any specific concern regarding this?

Reply:
Following a talk with Edi, we agreed I will investigate other options in other PR, as it is not crucial for this PR.

Reply:
> I don't know about performance cost, but when it comes to readability, I am a fan of subshell called on sourced functions.
> 
> It allows us to do something close to imports and calling functions from scopes. The alternative would be to either have one shell script per function or to keep a boilerplate to select the right function based on an attribute.
> 
> Is there any specific concern regarding this?

The issue is with sourcing. If you sub-shell, there is little reason to source.